types/next-auth/providers.d.ts

```diff
@@ -183,8 +183,8 @@ interface ProviderAuth0Options {
 type IdentityServer4 = (options: ProviderIS4Options) => GenericReturnConfig;
 
 interface ProviderIS4Options {
-    id: 'identity-server4';
-    name: 'IdentityServer4';
+    id: 'identity-server4' | string;
+    name: 'IdentityServer4' | string;
```

Comment:
As far as I understand, those two fields [are not customisable](https://github.com/nextauthjs/next-auth/blob/main/src/providers/identity-server4.js#L3-L4) 💅🏽  

Could you add some examples about your use case for this?

Reply:
I'm using this library right now
as I understood, id must be customizable because we can have multiple providers; here, we may have equal providers by different client-id and client-secrets but need to be unique by id.
Or even we can have custom providers as much as we want, they all need unique ids and name variables used in the default sign-in page.
So I think in this particle provider, we need the same aspect of having a custom id variable.

Reply:
Sorry, it's hard for me to understand what you're trying to explain 😅 , could you back it up with some code examples?

Or maybe @iaincollins you know what @raminrez refers to?

Reply:
as you can see on this page
https://next-auth.js.org/providers/identity-server4
These variables can be customized.

Reply:
> As far as I understand, those two fields are not customisable 💅🏽

Aren't they overridden by [this](https://github.com/nextauthjs/next-auth/blob/main/src/providers/identity-server4.js#L16)?

Reply:
yes, of course, they should be `string`
can I fix it now?

Reply:
Commit the new changes and I'll approve them

Reply:
thank you @JuanM04 ,new fix commited

Reply:
I didn't realise they were spreading the options at the bottom, I guess it makes sense then for it to be `string` 🤷🏽‍♂️ 👍🏽